rubix-spi/src/main/java/com/qubole/rubix/spi/fop/ObjectPoolPartition.java

```diff
@@ -78,85 +82,55 @@ public void returnObject(Poolable<T> object)
-        // Could not create objects, this is mostly due to connection timeouts hence no point blocking as there is not other producer of sockets
-        throw new RuntimeException("Could not add connections to pool");
-      }
-      // else wait for a connection to get free
+    try {
+      takeSemaphore.tryAcquire(config.getMaxWaitMilliseconds(), TimeUnit.MILLISECONDS);
+    }
+    catch (InterruptedException e) {
+      e.printStackTrace();
+      throw new RuntimeException("Cannot get a free object from the pool");
     }
 
-    Poolable<T> freeObject;
-    try {
-      if (blocking) {
-        freeObject = objectQueue.take();
+    Poolable<T> object;
+    if (blocking) {
+      try {
+        object = objectQueue.take();
       }
-      else {
-        freeObject = objectQueue.poll(config.getMaxWaitMilliseconds(), TimeUnit.MILLISECONDS);
-        if (freeObject == null) {
-          throw new RuntimeException("Cannot get a free object from the pool");
-        }
+      catch (InterruptedException e) {
+        throw new RuntimeException(e); // will never happen
       }
     }
-    catch (InterruptedException e) {
-      throw new RuntimeException(e); // will never happen
+    else
+    {
+      object = tryGetObject(objectQueue.poll());
     }
-
-    freeObject.setLastAccessTs(System.currentTimeMillis());
-    return freeObject;
+    object.setLastAccessTs(System.currentTimeMillis());
+    return object;
   }
 
-  private Poolable<T> increaseObjects(int delta, boolean returnObject)
+  private Poolable<T> tryGetObject(Poolable<T> poolable)
   {
-    int oldCount = totalCount;
-    if (delta + totalCount > config.getMaxSize()) {
-      delta = config.getMaxSize() - totalCount;
-    }
-
-    Poolable<T> objectToReturn = null;
-    try {
-      for (int i = 0; i < delta; i++) {
+    if (poolable == null)
+    {
+      try {
         T object = objectFactory.create(host, socketTimeout, connectTimeout);
         if (object != null) {
-          // Do not put the first object on queue
-          // it will be returned to the caller to ensure it's request is satisfied first if object is requested
-          Poolable<T> poolable = new Poolable<>(object, pool, host);
-          if (objectToReturn == null && returnObject) {
-            objectToReturn = poolable;
-          }
-          else {
-            objectQueue.put(poolable);
-          }
+          poolable = new Poolable<>(object, pool, host);
           totalCount++;
         }
-      }
-
-      if (delta > 0 && (totalCount - oldCount) == 0) {
-        log.warn(String.format("Could not increase pool size. Pool state: totalCount=%d queueSize=%d delta=%d", totalCount, objectQueue.size(), delta));
-      }
-      else {
         log.debug(String.format("Increased pool size by %d, to new size: %d, current queue size: %d, delta: %d",
-                totalCount - oldCount, totalCount, objectQueue.size(), delta));
+                1, totalCount, objectQueue.size(), 1));
```

Comment:
Remove constant `1`, change message to `Added a connection, new size: ..`

Reply:
yeah, this are draft changes, I will remove all these contents and redundant code in the next iteration.

Reply:
fixed.